### PR TITLE
Minimal twitter

### PR DESCRIPTION
**Pull request by Vretamoza**

### CODE REVIEW

File: lib/chirp/timeline.ex [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+    broadcast({:ok, post}, :post_updated)
+  end
+
+  def inc_reposts(%Post{id: id}) do
+    {1, [post]} =
+      from(p in Post, where: p.id == ^id, select: p)
+      |> Repo.update_all(inc: [reposts_count: 1])
+
+    broadcast({:ok, post}, :post_updated)
```
Comment (juanhurtado10) [inline lib/chirp/timeline.ex:37]: @Vretamoza could you please explain how `broadcast/2` works

File: lib/chirp/timeline/post.ex [hunk 2]
```diff
@@ -0,0 +1,21 @@
+defmodule Chirp.Timeline.Post do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "posts" do
+    field :body, :string
+    field :likes_count, :integer, default: 0
+    field :reposts_count, :integer, default: 0
+    field :username, :string, default: "Valery_Retamoza"
```
Comment (juanhurtado10) [inline lib/chirp/timeline/post.ex:9]: @Vretamoza why use `"Valery_Retamoza"` as a default username value?

File: lib/chirp_web/live/post_live/index.ex [hunk 3]
```diff
@@ -0,0 +1,57 @@
+defmodule ChirpWeb.PostLive.Index do
+  use ChirpWeb, :live_view
+
+  alias Chirp.Timeline
+  alias Chirp.Timeline.Post
+
+  @impl true
+  def mount(_params, _session, socket) do
```
Comment (juanhurtado10) [inline lib/chirp_web/live/post_live/index.ex:8]: @Vretamoza please explain `mount/3`

File: lib/chirp_web/live/post_live/form_component.ex [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    changeset = Timeline.change_post(post)
+
+    {:ok,
+     socket
+     |> assign(assigns)
+     |> assign(:changeset, changeset)}
+  end
+
+  @impl true
+  def handle_event("validate", %{"post" => post_params}, socket) do
```
Comment (juanhurtado10) [inline lib/chirp_web/live/post_live/form_component.ex:17]: @Vretamoza what is a `handle_event/3` in the LiveView context?

File: lib/chirp_web/live/post_live/form_component.ex [hunk 5]
```diff
@@ -0,0 +1,55 @@
+defmodule ChirpWeb.PostLive.FormComponent do
+  use ChirpWeb, :live_component
+
+  alias Chirp.Timeline
+
+  @impl true
+  def update(%{post: post} = assigns, socket) do
```
Comment (juanhurtado10) [inline lib/chirp_web/live/post_live/form_component.ex:7]: @Vretamoza why not use a `mount/3` instead?

File: lib/chirp_web/live/post_live/form_component.html.heex [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<div>
+  <h2><%= @title %></h2>
+
+  <.form
```
Comment (juanhurtado10) [inline lib/chirp_web/live/post_live/form_component.html.heex:4]: @Vretamoza did you fully understand the difference between `leex` and `heex` engines?

File: lib/chirp_web/live/post_live/show.ex [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+  alias Chirp.Timeline
+
+  @impl true
+  def mount(_params, _session, socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def handle_params(%{"id" => id}, _, socket) do
```
Comment (juanhurtado10) [inline lib/chirp_web/live/post_live/show.ex:12]: @Vretamoza what is the difference between `handle_event` and `handle_params`?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: lib/chirp/timeline.ex (hunk 1 above)
Comment (Vretamoza) [reply lib/chirp/timeline.ex:37]: Broadcast is a function created to send the changes made to all the users subscribed to a topic, using PubSub.

File: lib/chirp/timeline/post.ex (hunk 2 above)
Comment (Vretamoza) [reply lib/chirp/timeline/post.ex:9]: In order to be able to create posts with that username, as the form doesn't ask for it, just for a post body.

File: lib/chirp_web/live/post_live/index.ex (hunk 3 above)
Comment (Vretamoza) [reply lib/chirp_web/live/post_live/index.ex:8]: It's the function that connects all the socket assigns needed to render the view. It is called twice, to first load the page and then to set up the live socket connection.

File: lib/chirp_web/live/post_live/form_component.ex (hunk 4 above)
Comment (Vretamoza) [reply lib/chirp_web/live/post_live/form_component.ex:17]: It handles the bindings in LiveView, bindings are events send by the client, like clicks to buttons or pressed keys.

File: lib/chirp_web/live/post_live/show.ex (hunk 7 above)
Comment (Vretamoza) [reply lib/chirp_web/live/post_live/show.ex:12]: handle_event is a function that allows us to handle events that come from the client, handle_params allows us to load data once there is a LiveView connection, it is called after the mount function.

File: lib/chirp_web/live/post_live/form_component.html.heex (hunk 6 above)
Comment (Vretamoza) [reply lib/chirp_web/live/post_live/form_component.html.heex:4]: I know the interpolation sintax changed and the sigil used to write the function component changed from ~L to ~H.

File: lib/chirp_web/live/post_live/form_component.ex (hunk 5 above)
Comment (Vretamoza) [reply lib/chirp_web/live/post_live/form_component.ex:7]: mount is just called when the component is first added to the page, in this case, we want to update a post body, so we call the update function that receives the assigns given to live_component/1

**Pull request merged**